examples/azure/terraform-azure-nsg-example/README.md

```diff
@@ -0,0 +1,129 @@
+1. `go build terraform_azure_nsg_example_test.go`
+1. `go test -v -run TestTerraformAzureNsgExample`
+
+## Module test APIs
+
+This module exposes the following top-level API methods:
+
+- GetDefaultNsgRulesClientE returns a rules client that retuns the _default_ NSG rules<br>
+`func GetDefaultNsgRulesClientE(subscriptionID string) (network.DefaultSecurityRulesClient, error)`
+- GetCustomNsgRulesClientE returns a rules client that returns the _custom_ NSG rules<br>
+`func GetCustomNsgRulesClientE(subscriptionID string) (network.SecurityRulesClient, error)`
+- GetAllNSGRulesE returns a slice containing all rules from the NSG (including defauls and custom)<br>
+`func GetAllNSGRulesE(resourceGroupName, nsgName, subscriptionID string) (NsgRuleSummaryList, error)`
+
+Note that the `GetAllNsgRulesE` method returns an `NsgRuleSummaryList`, which is defined as follows:
+
+- NsgRuleSummaryList holds a colleciton of NsgRuleSummary structs<br>
+` type NsgRuleSummaryList struct { SummarizedRules []NsgRuleSummary } `
+
+This collection wrapper has the following method:
+
+- FindRuleByName looks for a matching rule name <br>
+`func (summarizedRules *NsgRuleSummaryList) FindRuleByName(t *testing.T, name string) NsgRuleSummary`
+
+Finally, the inner collection is defined as an `NsgRuleSummary` struct, defined as follows:
+
+```
+type NsgRuleSummary struct {
+	Name                     string
+	Description              string
+	Protocol                 string
+	SourcePortRange          string
+	DestinationPortRange     string
+	SourceAddressPrefix      string
+	DestinationAddressPrefix string
+	Access                   string
+	Priority                 int32
+	Direction                string
+}
+```
+
+This wrapper has the following methods available:
+
+- AllowsDestinationPort checks to see if the rule allows a specific destination port<br>
+`func (summarizedRule *NsgRuleSummary) AllowsDestinationPort(port string) bool`
+- AllowsSourcePort checks to see if the rule allows a specific source port<br>
+`func (summarizedRule *NsgRuleSummary) AllowsSourcePort(port string) bool`
+
+## Check Go Dependencies
+
+Check that the `github.com/Azure/azure-sdk-for-go` version in your generated `go.mod` for this test matches the version in the terratest [go.mod](https://github.com/gruntwork-io/terratest/blob/master/go.mod) file.  
+
+> This was tested with **go1.14.4**.
+
+### Check Azure-sdk-for-go version
+
+Let's make sure [go.mod](https://github.com/gruntwork-io/terratest/blob/master/go.mod) includes the appropriate [azure-sdk-for-go version](https://github.com/Azure/azure-sdk-for-go/releases/tag/v46.0.0):
+
+```go
+require (
+    ...
+    github.com/Azure/azure-sdk-for-go v46.0.0+incompatible
+    ...
+)
+```
+
+If we make changes to either the **go.mod** or the **go test file**, we should make sure that the go build command works still.
+
+```powershell
+go build terraform_azure_nsg_example_test.go
+```
+
+## Review Environment Variables
```

Comment:
Same suggestion as in https://github.com/gruntwork-io/terratest/pull/626/files#r490158239 to extract this to a common location and cross-link.

Reply:
See PR #626  - common `README` will be available there.